docs/user_guide.md

```diff
@@ -0,0 +1,37 @@
+# Using S2I images
```

Comment:
i think this needs an intro paragraph.  who's the audience, what am i going to learn.  maybe a table of context too.  jumping right into "invoking scripts embedded into an image" isn't a gentle enough entrypoint to this topic.